src/components/OptionsBar/Options/FilterModal/Filters/RelativeFilters.jsx

```diff
@@ -0,0 +1,48 @@
+import { useSelector } from 'react-redux';
+import { useRef } from 'react';
+
+const relativeFilters = (filterIds, tabIndex) => {
+  const filters = useSelector((state) => state.filters);
+
+  const getFilterMatrix = () => {
+    const filterMatrix = new Array(filterIds.length);
+
+    filterIds.forEach((id, index) => {
+      filterMatrix[index] = filters[id];
+    });
+
+    return filterMatrix;
+  };
+
+  const queuedFilters = useRef(getFilterMatrix());
+
+  const getFilters = () => queuedFilters.current;
+
+  const getRelativeFilters = () => getFilters()[tabIndex];
+
+  const pushRelativeFilters = (filter) => getRelativeFilters().push(filter);
+
+  const spliceRelativeFiltersAt = (index) => getRelativeFilters().splice(index, 1);
```

Comment:
Here you have two functions, push and splice. push returns the new length and splice the removed elements. It is a bit conflicting, as they act as opposite of one another, yet they have entirely different return values. For a more cohesive code, I would return the resulting array